api/src/main/java/io/grpc/ManagedChannelRegistry.java

```diff
@@ -144,6 +151,31 @@ static List<Class<?>> getHardCodedClasses() {
   }
 
   ManagedChannelBuilder<?> newChannelBuilder(String target, ChannelCredentials creds) {
+    return newChannelBuilder(NameResolverRegistry.getDefaultRegistry(), target, creds);
+  }
+
+  @VisibleForTesting
+  ManagedChannelBuilder<?> newChannelBuilder(NameResolverRegistry nameResolverRegistry,
+      String target, ChannelCredentials creds) {
+    URI uri;
+    try {
+      uri = new URI(target);
+    } catch (URISyntaxException e) {
+      throw new ProviderNotFoundException(e.getMessage());
+    }
+    String scheme = uri.getScheme();
+    if (scheme == null) {
```

Comment:
Okay this makes sense and may be why we should ignore `URISyntaxException`. But I was thinking of a test string like `":test123"` which generates `URISyntaxException` and giving it to the defaultScheme's nameResolver is not going to be of any use (I think). So it will just fail later inside the channel code. What do you think of this case?